src/background/controllers/accountController.ts

```diff
@@ -431,80 +417,77 @@ export default class AccountController extends IController {
-        break;
-      case MESSAGE_TYPE.GET_WALLET_INFO:
-        sendResponse(this.loggedInAccount && this.loggedInAccount.wallet
-          ? this.loggedInAccount.wallet.info : undefined);
-        break;
-      case MESSAGE_TYPE.GET_QTUM_USD:
-        sendResponse(this.loggedInAccount && this.loggedInAccount.wallet
-          ? this.loggedInAccount.wallet.qtumUSD : undefined);
-        break;
-      case MESSAGE_TYPE.VALIDATE_WALLET_NAME:
-        sendResponse(this.isWalletNameTaken(request.name));
-        break;
-      case MESSAGE_TYPE.GET_MAX_QTUM_SEND:
-        this.sendMaxQtumAmountToPopup();
-        break;
-      default:
-        break;
+    try {
+      switch (request.type) {
+        case MESSAGE_TYPE.LOGIN:
+          this.login(request.password);
+          break;
+        case MESSAGE_TYPE.IMPORT_MNEMONIC:
+          this.importMnemonic(request.accountName, request.mnemonicPrivateKey);
+          break;
+        case MESSAGE_TYPE.IMPORT_PRIVATE_KEY:
+          this.importPrivateKey(request.accountName, request.mnemonicPrivateKey);
+          break;
+        case MESSAGE_TYPE.SAVE_TO_FILE:
+          this.saveToFile(request.accountName, request.mnemonicPrivateKey);
+          break;
+        case MESSAGE_TYPE.ACCOUNT_LOGIN:
+          this.loginAccount(request.selectedWalletName);
+          break;
+        case MESSAGE_TYPE.SEND_TOKENS:
+          this.sendTokens(request.receiverAddress, request.amount, request.transactionSpeed);
+          break;
+        case MESSAGE_TYPE.LOGOUT:
+          this.logoutAccount();
+          break;
+        case MESSAGE_TYPE.HAS_ACCOUNTS:
+          sendResponse(this.hasAccounts);
+          break;
+        case MESSAGE_TYPE.GET_ACCOUNTS:
+          sendResponse(this.accounts);
+          break;
+        case MESSAGE_TYPE.GET_LOGGED_IN_ACCOUNT:
+          sendResponse(this.loggedInAccount && this.loggedInAccount.wallet && this.loggedInAccount.wallet.info
+            ? { name: this.loggedInAccount.name, address: this.loggedInAccount!.wallet!.info!.addrStr }
+            : undefined);
+          break;
+        case MESSAGE_TYPE.GET_LOGGED_IN_ACCOUNT_NAME:
+          sendResponse(this.loggedInAccount ? this.loggedInAccount.name : undefined);
+          break;
+        case MESSAGE_TYPE.GET_WALLET_INFO:
+          sendResponse(this.loggedInAccount && this.loggedInAccount.wallet
+            ? this.loggedInAccount.wallet.info : undefined);
+          break;
+        case MESSAGE_TYPE.GET_QTUM_USD:
+          sendResponse(this.loggedInAccount && this.loggedInAccount.wallet
+            ? this.loggedInAccount.wallet.qtumUSD : undefined);
+          break;
+        case MESSAGE_TYPE.VALIDATE_WALLET_NAME:
+          sendResponse(this.isWalletNameTaken(request.name));
+          break;
+        case MESSAGE_TYPE.GET_MAX_QTUM_SEND:
+          this.sendMaxQtumAmountToPopup();
+          break;
+        default:
+          break;
+      }
+    } catch (err) {
```

Comment:
:+1: This is the proper place for this. Each layer should handle its own error handling: bg, UI, etc.